src/components/Form/index.tsx

```diff
@@ -0,0 +1,171 @@
+import * as React from 'react';
+
+import type {
+    Comment,
+    CommentReply,
+} from '../../state/comments';
+
+
+export interface CommentFormSetProps {
+    comments: Comment[];
+}
+
+export function CommentFormSetComponent(props: CommentFormSetProps) {
+    const [initialNumber] = React.useState(props.comments.filter(comment => {return comment.remoteId != null}).length);
+    const prefix = "comments";
+
+    const commentForms = props.comments.map((comment, formNumber) => (
+        <CommentFormComponent
+            key={comment.localId}
+            comment={comment}
+            formNumber={formNumber}
+        />
+    ));
+
+    return (<>
+        <PrefixedHiddenInput
+            fieldName="TOTAL_FORMS"
+            value={props.comments.length}
+            prefix={prefix}
+        />
+        <PrefixedHiddenInput
+            fieldName="INITIAL_FORMS"
+            value={initialNumber}
+            prefix={prefix}
+        />
+        <PrefixedHiddenInput
+            fieldName="MIN_NUM_FORMS"
+            value="0"
+            prefix={prefix}
+        />
+        <PrefixedHiddenInput
+            fieldName="MAX_NUM_FORMS"
+            value=""
+            prefix={prefix}
+        />
+        {commentForms}
+    </>)
+}
+
+interface PrefixedHiddenInputProps {
+    prefix: string;
+    value: number|string;
+    fieldName: string;
+}
+
+function PrefixedHiddenInput(props: PrefixedHiddenInputProps) {
+    return (<input
```

Comment:
**Nitpick:** Don't we have a custom components for `<input`>? It's generally good to have one, because then you may need an exercise of e.g. assigning the same `className` trillion times in your codebase. Or adding some other global behaviour.

Reply:
A general purpose `<input>` react component? Can't see one in Wagtail at the moment, and there aren't any other plain `<input>` elements in commenting right now (textarea and button only). I'm not averse to it, though I can't see a big advantage right now since all inputs are `PrefixedHiddenInput`. Would you suggest adding a placeholder one anyway, even if it's basically just the plain html `<input>`?

Reply:
True, it's hidden input... For visual ones it may be of advantage if we ever add any styling to it or some common validation things, but it's good as it is for now.